apollo-router/src/plugins/demand_control/cost_calculator/static_cost.rs

```diff
@@ -1073,4 +1129,25 @@ mod tests {
         assert_eq!(planned_cost_rust(schema, query, variables), 127.0);
         assert_eq!(actual_cost(schema, query, variables, response), 125.0);
     }
+
+    #[test]
+    fn arbitrary_json_as_custom_scalar_in_variables() {
+        let schema = include_str!("./fixtures/arbitrary_json_schema.graphql");
+        let query = r#"
+            query FetchData($myJsonValue: ArbitraryJson) {
+                fetch(args: {
+                    json: $myJsonValue
+                })
+            }
+        "#;
+        let variables = r#"
+            {
+                "myJsonValue": {
+                    "field.with.dots": 1
+                }
+            }
```

Comment:
the idea is not to let demand control scoring happen via this arbitrary json, right? or is that also the idea?

Reply:
The idea is that even though it is a JSON structure, it should be scored as a scalar in the same way we treat built-in scalars like `Int` or `String` (free, unless annotated with `@cost`). Previously, I was naively assuming that any JSON object structure passed in as variables could be converted into the equivalent AST structure. Unfortunately the custom scalar pattern means that the otherwise compliant JSON structure could contain a subtree of non-compliant JSON.